### PR TITLE
Add welcoming message

### LINKED ISSUE #10: Improve onboarding dialog
We would need a link to the GDPR policy, a welcome message explaining possible features, and talking a bit about xtream as the company behind the scene

### PR DESCRIPTION
**Pull request by simone3991**
closes #10

### CODE REVIEW

File: src/Supports/BotHandler.cs [hunk 1]
```diff
@@ -2,11 +2,14 @@
 using System.Threading.Tasks;
 using Bot.States;
 using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
 
 namespace Bot.Supports
 {
     public interface IBotHandler
```
Comment (khaelys) [inline src/Supports/BotHandler.cs:9]: If we have an handler that is not a dialog we break LSP. I guess that we will need two types of chain in that case at this point. 

I propose to just document this interfaces and assume for now that  it should be used only for dialogs. Think about renaming this interface to something like `IDialogHandler` or `IBotDialogHandler`

File: src/Supports/UtilityHandler.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    public class UtilityHandler : IBotHandler
+    {
+        private readonly LuisRecognizerProxy _luisRecognizer;
+        private readonly DialogSet _dialogSet;
+
+        public UtilityHandler(ConversationState conversationState, LuisRecognizerProxy luisRecognizer)
+        {
+            IStatePropertyAccessor<DialogState> dialogState = conversationState.CreateProperty<DialogState>("UtilityDialogState");
+            _luisRecognizer = luisRecognizer;
+            _dialogSet = new DialogSet(dialogState);
```
Comment (khaelys) [inline src/Supports/UtilityHandler.cs:20]: In this case we have no dialog or am I missing something? How a dialog can be active if this set is empty?

**Review by khaelys — CHANGES_REQUESTED** [review]

File: src/Supports/UtilityHandler.cs (hunk 2 above)
Comment (simone3991) [reply src/Supports/UtilityHandler.cs:20]: it doesn't trigger any multistep dialog, it just sends message activities that are self-conclusive

File: src/Supports/BotHandler.cs (hunk 1 above)
Comment (simone3991) [reply src/Supports/BotHandler.cs:9]: what do you mean an handler that is not a dialog? I thought handlers are what they are called - partial functions to handle incominc activities. Such handling could reply with a text, or trigger a dialog, or even multiple dialogs

if anything it's more of a conversation handler, in that it manages the flow of conversation delegating to chain nodes or taking priority over the chain when an handler is in the middle of a dialog

File: src/Clockify/ClockifyHandler.cs [hunk 3]
```diff
@@ -72,50 +66,27 @@ public class ClockifyHandler : IBotHandler
...
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(topIntent), topIntent, null);
+                    return true;
+                default: 
+                    return false;
             }
+        }
 
-            return false;
+        public DialogSet GetDialogSet()
```
[the remark this answers is not in the document]
Comment (simone3991) [reply src/Clockify/ClockifyHandler.cs:85]: you mean like this?
`public DialogSet GetDialogSet() => _dialogSet;`

File: src/Supports/BotHandler.cs (hunk 1 above)
Comment (khaelys) [reply src/Supports/BotHandler.cs:9]: it's just strange that we should create an empty dialog in the utility handler, we can proceed this way, but we should document it properly in the interface

**Pull request merged**